### DISCUSSION TITLE
Can you remove "| null" from model relations?

### DISCUSSION THREAD

**Discussion by drakedeatonuk** [prisma/prisma#25596] [Q&A]
### Question

Currently all of my prisma model relations return "| null" when querying tables.

For example, given the below schema:

```
model User {
  userId        Int          @id @default(autoincrement())

  address Address?
}

model Address {
  addressId Int              @id @default(autoincrement())

  user User @relation(fields: [userId], references: [userId])
}
```

a simple function call like this:
```
db.address.find({
  where: { userId: 1 }
  include: { address: true }
```

returns the relation `{ address: Address | null }`.

this is pretty much never really useful for me as these relationships always exist.

Is there any way to change this?

Thanks in advance!

**Comment by nurul3101** [comment]
Hello @drakedeatonuk 👋 

You are getting `| null` in the generated types because Address is marked as optional in the `User` model.

As mentioned in our docs [here](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/one-to-one-relations#required-and-optional-1-1-relation-fields), In a one-to-one relation the side of the relation without a relation scalar (the field representing the foreign key in the database) **must** be optional. Prisma doesn't allow required relations on the side without the foreign key, as it can't be enforced at the database level.

If you're certain that the relation always exists, you can use a non-null assertion in TypeScript when querying:

```ts
const user = await prisma.user.findUnique({
  where: { userId: 1 },
  include: { address: true },
});

const address = user.address!; // Use non-null assertion
```

**Comment by drakedeatonuk** [comment]
Hi @nurul3101, thanks for the info.

I get the reason why it works that way but it's a hinderance in 99% of my use cases.

Using type assertions is a bit of a code smell, particularly when you've got large complex queries.

It would be great if this could just be baked into the API, eg:

```
const user = await prisma.user.findUnique({
  where: { userId: 1 },
  include: { 
    address: {
      enforce: {
        isNot: null
      }
    }
  },
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
